index.js

```diff
@@ -117,7 +117,7 @@ function generateRewritePrefix (prefix, opts) {
     return ''
   }
 
-  let rewritePrefix = opts.rewritePrefix || new URL(opts.upstream).pathname
+  let rewritePrefix = opts.rewritePrefix || new URL(opts.upstream || 'http://empty').pathname
```

Comment:
Isn't this `http://empty` default value avoidable?

Reply:
I think so, we should just use a default pathname if no upstream is there.

Reply:
We need to provide any valid upstream to URL constructor, in another case, we'll have to handle the exception that the URL constructor will throw.

Reply:
I mean if there is no `opts.upstream`, jut specify `/` as the `rewritePrefix`. We can skip parsing the URL.

Reply:
`new URL('http://empty').pathname` this construction should return `/`. 
Another way to solve this situation is to additionally check upstream and code will be looking like that: 
`let rewritePrefix = opts.rewritePrefix || opts.upstream ? new URL(opts.upstream).pathname : '/'`

Reply:
So if the second option is preferred I can update that part.

Reply:
To me it looks more readable

Reply:
I have refactored rewritePrefix as requested.